user.go

```diff
@@ -91,17 +106,41 @@ func (u *User) Mention() string {
 }
 
 // AvatarURL returns a URL to the user's avatar.
-//    size:    The size of the user's avatar as a power of two
-//             if size is an empty string, no size parameter will
-//             be added to the URL.
+//
+//	size:    The size of the user's avatar as a power of two
+//	         if size is an empty string, no size parameter will
+//	         be added to the URL.
 func (u *User) AvatarURL(size string) string {
-	return avatarURL(u.Avatar, EndpointDefaultUserAvatar(u.Discriminator),
-		EndpointUserAvatar(u.ID, u.Avatar), EndpointUserAvatarAnimated(u.ID, u.Avatar), size)
+	return avatarURL(
+		u.Avatar,
+		EndpointDefaultUserAvatar(u.AvatarIndex()),
+		EndpointUserAvatar(u.ID, u.Avatar),
+		EndpointUserAvatarAnimated(u.ID, u.Avatar),
+		size,
+	)
 }
 
 // BannerURL returns the URL of the users's banner image.
-//    size:    The size of the desired banner image as a power of two
-//             Image size can be any power of two between 16 and 4096.
+//
+//	size:    The size of the desired banner image as a power of two
+//	         Image size can be any power of two between 16 and 4096.
 func (u *User) BannerURL(size string) string {
 	return bannerURL(u.Banner, EndpointUserBanner(u.ID, u.Banner), EndpointUserBannerAnimated(u.ID, u.Banner), size)
 }
+
+// isMigrated returns true if the user is migrated from the legacy username system
+// https://discord.com/developers/docs/change-log#identifying-migrated-users
+func (u *User) isMigrated() bool {
+	return u.Discriminator == "" || u.Discriminator == "0"
+}
+
+// AvatarIndex returns the index of the user's avatar
+func (u *User) AvatarIndex() uint64 {
+	if u.isMigrated() {
+		id, _ := strconv.ParseUint(u.ID, 10, 64)
+		return (id >> 22) % 6
+	}
+
+	id, _ := strconv.Atoi(u.Discriminator)
+	return uint64(id % 5)
+}
```

Comment:
Probably better to name it `DefaultAvatarIndex`, as this can be confused with normal avatar.